h/static/scripts/controllers/tooltip-controller.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/**
+ * A custom tooltip similar to the one used in Google Docs which appears
+ * instantly when activated on a target element.
+ *
+ * The tooltip is displayed and hidden by setting its target element.
+ *
+ *  var tooltip = new Tooltip(document.body);
+ *  tooltip.setState({target: aWidget}); // Show tooltip
+ *  tooltip.setState({target: null}); // Hide tooltip
+ *
+ * The tooltip's label is derived from the target element's 'aria-label'
+ * attribute.
+ *
+ * @param {Element} el - The container for the tooltip.
+ */
+function TooltipController(el) {
```

Comment:
Once https://github.com/hypothesis/h/pull/3771 lands this should be refactored to use the `Controller` base class.